### PR TITLE
Attempt to support v2 and v1 at the same time where possible

### PR DESCRIPTION
**Pull request by simensen**
Since the testing of this package and the testing subsplit package themselves define classes that are incompatible with eventsauce/eventsauce:v2, the top-level package requires ^2.0@dev and the testing-related subsplit packages requires ^2.0.

Otherwise, it *does* seem like the implementations of these subsplit packages should continue to work with v1 without any changes.

----

Unclear if it would make sense to bump from `0.1` to `0.2` for these changes. If that's the case, there is a bit more work for me to do here. I think this should all work as-is but would love to figure out if this should go into its own branch separate from the current 0.1 line.

In short: 0.1 could be support for v1 and 0.2 could be support for both v1 and v2. If you agree and like that idea, I can send another PR with branch aliases for the top-level project and the subsplit projects.

### CODE REVIEW

**Comment by frankdejonge** [comment]
@simensen I might be mistaken, but if we declare the return type of the fromPayload to `static` won't this be backwards compatible?

**Comment by simensen** [comment]
@frankdejonge unless I'm misunderstanding,  changing return type to static is a BC break from PHP perspective. PHP itself threw errors about incompatible return type if I did not explicitly return `static` in the implementations.

The rest of the project (except concrete implementations of aggregate root ID and serializable payload which seem to be only for tests or actual exported test tooling) seems to work just fine with both versions of the core eventsauce package (1.x or 2.x). So, it's only the testing of the root of the mono repo and the testing-specific subsplit packages that need to be locked to either 1.x or 2.x as the implementations of the interfaces are not compatible.

Also, looks like this PR needs a rebase.  I'll hold off on additional work until we're on the same page here. Entirely possible I'm off on something here and if we CAN make it compatible with both versions of the core package that would be awesome. :)

**Comment by simensen** [comment]
Oh, hey, so, when I'm wrong, I'm wrong! You were right. :) I've loosened the restrictions to include both v1 and v2 as it looks like the `static` return type on implementations that originally required a specific type returned is not an issue. Where I *did* have issues was the reverse: once the interface returns static, the implementations are required to also return static.

 * https://3v4l.org/frGri – Example using 2.x signature w/ implementation returning interface name
 * https://3v4l.org/jaSu4 – Example using 1.x signature w/ implementation returning `static`

**TL;DR** I've updated this PR to allow for both 2.x and 1.x. Currently, the testing setup will probably default to just working with 1.x until there is a stable 2.x release of `eventsauce/eventsauce`.

Both the root package and all subsplit packages now require `^1.0||^2.0`. 🎉
